pkg/ruler/mapper.go

```diff
@@ -136,7 +138,7 @@ func (m *mapper) MapRules(user string, ruleConfigs map[string][]rulefmt.RuleGrou
 	return anyUpdated, filenames, nil
 }
 
-func (m *mapper) writeRuleGroupsIfNewer(groups []rulefmt.RuleGroup, filename string) (bool, error) {
+func (m *mapper) writeRuleGroupsIfNewer(groups []rulefmt.RuleGroup, filename string, logger log.Logger) (bool, error) {
```

Comment:
You might want to add a comment or something explaining that we use this logger instead of `m.logger` directly because we want to record a `userID`. Otherwise, I think at first glance, this will look redundant to someone without context.